mcnp/mcnp_funcs.cpp

```diff
@@ -476,16 +476,38 @@ void write_lcad_old(std::ofstream &lcadfile)
   int num_surfs = DAG->num_entities( 2 );
 
   int cmat = 0;
-  double crho, cimp = 1.0;
+  double crho = 0;
+  double cimp_n = 0, cimp_p = 0, cimp_e = 0;
+
+  // Detect which importances are used so all cells, including implicit
+  // complement and graveyard, have these importances
+  bool imp_n_needed = false, imp_p_needed = false, imp_e_needed = false;
+  for( int i = 1; i <= num_cells; ++i ) {
+    moab::EntityHandle vol = DAG->entity_by_index( 3, i );
+    if( DAG->has_prop( vol, "imp.n" )) {
```

Comment:
I would like to change the way that metadata is handled by Dagmc in general, for example at loading store it all in some metadata class so we can query it more neartly. But right now, one has use has_prop.